aws/resource_aws_codedeploy_deployment_group.go

```diff
@@ -630,6 +896,85 @@ func alarmConfigToMap(config *codedeploy.AlarmConfiguration) []map[string]interf
 	return result
 }
 
+// deploymentStyleToMap converts a codedeploy.DeploymentStyle object
+// into a []map[string]interface{} list containing a single item
+func deploymentStyleToMap(style *codedeploy.DeploymentStyle) []map[string]interface{} {
+	result := make([]map[string]interface{}, 0, 1)
+
+	if style == nil {
+		return result
+	}
+
+	item := make(map[string]interface{})
+	item["deployment_option"] = *style.DeploymentOption
+	item["deployment_type"] = *style.DeploymentType
```

Comment:
Are you sure that none of these options will ever be `nil`? Otherwise this would cause a crash here.